vimiv/imutils/exif.py

```diff
@@ -10,139 +10,361 @@
+
+    def __init__(self, filename=""):
+        self._metadata = None
+
+        try:
+            self._metadata = piexif.load(filename)
+        except FileNotFoundError:
+            _logger.debug("File %s not found", filename)
+            return
 
-@check_piexif("")
-def exif_date_time(filename) -> str:
-    """Exif creation date and time of filename."""
-    with contextlib.suppress(piexif.InvalidImageDataError, FileNotFoundError, KeyError):
-        exif_dict = piexif.load(filename)
-        return exif_dict["0th"][piexif.ImageIFD.DateTime].decode()
-    return ""
+    def get_formatted_exif(self) -> Dict[str, str]:
 
+        desired_keys = [
+            e.strip() for e in api.settings.metadata.current_keyset.value.split(",")
+        ]
+        _logger.debug(f"Read metadata.current_keys {desired_keys}")
 
-class ExifInformation(dict):
-    """Dictionary to load and store exif information of a single image.
+        exif = dict()
 
-    Exif information is loaded and formatted upon construction. Afterwards the class
-    behaves like a regular dictionary.
+        try:
 
-    Attributes:
-        _exif: Loaded piexif exif information dictionary if any.
+            for ifd in self._metadata:
+                if ifd == "thumbnail":
+                    continue
+
+                for tag in self._metadata[ifd]:
+                    keyname = piexif.TAGS[ifd][tag]["name"]
+                    keytype = piexif.TAGS[ifd][tag]["type"]
+                    val = self._metadata[ifd][tag]
+                    _logger.debug(
+                        f"name: {keyname}\
+                        type: {keytype}\
+                        value: {val}\
+                        tag: {tag}"
+                    )
+                    if keyname not in desired_keys:
+                        _logger.debug(f"Ignoring key {keyname}")
+                        continue
+                    if keytype in (
+                        piexif.TYPES.Byte,
+                        piexif.TYPES.Short,
+                        piexif.TYPES.Long,
+                        piexif.TYPES.SByte,
+                        piexif.TYPES.SShort,
+                        piexif.TYPES.SLong,
+                        piexif.TYPES.Float,
+                        piexif.TYPES.DFloat,
+                    ):  # integer and float
+                        exif[keyname] = (keyname, val)
+                    elif keytype in (
+                        piexif.TYPES.Ascii,
+                        piexif.TYPES.Undefined,
+                    ):  # byte encoded
+                        exif[keyname] = (keyname, val.decode())
+                    elif keytype in (
+                        piexif.TYPES.Rational,
+                        piexif.TYPES.SRational,
+                    ):  # (int, int) <=> numerator, denominator
+                        exif[keyname] = (keyname, f"{val[0]}/{val[1]}")
+
+        except (piexif.InvalidImageDataError, KeyError):
+            return None
```

Comment:
I think we should just return an empty dict here, i.e. `return {}`, as we do not deal with `None` explicitly and an empty dict is falsy in itself.

Reply:
Thanks for pointing this out. I have fixed it in 7ef003c7dd9f43f36816598bacf6b0fe643ed368